analyzers/src/SonarAnalyzer.CSharp/Rules/CommentedOutCode.cs

```diff
@@ -34,163 +34,115 @@ namespace SonarAnalyzer.Rules.CSharp
+        private static void CheckMultilineComment(SyntaxTreeAnalysisContext context, SyntaxTrivia trivia)
         {
-            var triviaContent = GetTriviaContent(comment);
-            var triviaLines = triviaContent.Split(MetricsBase.LineTerminators, StringSplitOptions.None);
+            var triviaLines = TriviaContent().Split(MetricsBase.LineTerminators, StringSplitOptions.None);
 
             for (var triviaLineNumber = 0; triviaLineNumber < triviaLines.Length; triviaLineNumber++)
             {
-                if (!IsCode(triviaLines[triviaLineNumber]))
-                {
-                    continue;
-                }
-
-                var triviaStartingLineNumber = comment.GetLocation().GetLineSpan().StartLinePosition.Line;
-                var lineNumber = triviaStartingLineNumber + triviaLineNumber;
-                var lineSpan = context.Tree.GetText().Lines[lineNumber].Span;
-                var commentLineSpan = lineSpan.Intersection(comment.GetLocation().SourceSpan);
-
-                var location = Location.Create(context.Tree, commentLineSpan ?? lineSpan);
-                context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, location));
-                return;
-            }
-        }
-
-        private static string GetTriviaContent(SyntaxTrivia trivia)
-        {
-            var triviaContent = trivia.ToString();
-            if (trivia.IsKind(SyntaxKind.MultiLineCommentTrivia))
-            {
-                if (triviaContent.StartsWith("/*", StringComparison.Ordinal))
+                if (IsCode(triviaLines[triviaLineNumber]))
                 {
-                    triviaContent = triviaContent.Substring(2);
+                    var triviaStartingLineNumber = trivia.GetLocation().GetLineSpan().StartLinePosition.Line;
+                    var lineNumber = triviaStartingLineNumber + triviaLineNumber;
+                    var lineSpan = context.Tree.GetText().Lines[lineNumber].Span;
+                    var commentLineSpan = lineSpan.Intersection(trivia.GetLocation().SourceSpan);
+                    var location = Location.Create(context.Tree, commentLineSpan ?? lineSpan);
+                    context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, location));
+                    return;
                 }
-
-                if (triviaContent.EndsWith("*/", StringComparison.Ordinal))
-                {
-                    triviaContent = triviaContent.Substring(0, triviaContent.Length-2);
-                }
-                return triviaContent;
             }
 
-            if (trivia.IsKind(SyntaxKind.SingleLineCommentTrivia))
+            string TriviaContent()
             {
-                if (triviaContent.StartsWith("//", StringComparison.Ordinal))
-                {
-                    triviaContent = triviaContent.Substring(2);
-                }
-
-                return triviaContent;
+                var content = trivia.ToString().Substring(CommentMarkLength);
```

Comment:
_About these coverage changes_, Roslyn is directly going for the substrings as well, when comment kind is present
https://github.com/dotnet/roslyn/blob/main/src/Analyzers/CSharp/Analyzers/FileHeaders/CSharpFileHeaderHelper.cs#L28